### PR TITLE
Disconnected Environment

### PR DESCRIPTION
**Pull request by spitzerr**
Support Configuration of ArcGIS Enterprise for Disconnected Environments according to:

http://enterprise.arcgis.com/en/portal/latest/administer/windows/configure-a-disconnected-deployment.htm

### CODE REVIEW

**Comment by shailesh91** [comment]
@spitzerr I think instead of adding the whole config.js inside the config JSON file, I think we should instead give a path to config.js file.

**Comment by shailesh91** [comment]
@spitzerr I think we should abstract the whole disconnected envirornments workflow into a separate Configuration (as you suggested earlier) and run it as post deployment operation {DisconnectedEnv}.  The reason is that we might need to publish some services which is esentially a post deployment operation. 
```Configure-ArcGIS [[Path to Configuration json File]] -Mode [DisconnectedEnv] ```

**Comment by spitzerr** [comment]
@shailesh91 Totally agree on post-deployment scenario...

**Comment by spitzerr** [comment]
@shailesh91 changed config.js to file Copy instead of Property-Setting in https://github.com/Esri/arcgis-powershell-dsc/pull/43/commits/47afbbb0baeebfe56f24b4bd8367542b511a491c

**Comment by shailesh91** [comment]
Basemaps in a disconnected environment
1. Needs an accessible Tile Service hosted on a local server.
2. Create a group
https://developers.arcgis.com/rest/users-groups-and-items/create-group.htm
3. Create a base map 
https://developers.arcgis.com/rest/users-groups-and-items/add-item.htm
https://developers.arcgis.com/web-map-specification/objects/baseMap/
https://developers.arcgis.com/python/sample-notebooks/publishing-web-maps-and-web-scenes/
4. Share the base map
5. Set the Basemap Gallery and Default Basemap

Will expand on this more as I find out more.

**Comment by shailesh91** [comment]
@spitzerr Commit 890c00b should fix this error.

File: Modules/ArcGIS/DSCResources/ArcGIS_Portal_DisconnectedEnvironment/ArcGIS_Portal_DisconnectedEnvironment.psm1 [hunk 1]
```diff
@@ -0,0 +1,1042 @@
...
+                    if (-not ($CurHelperServices.geometry.url.StartsWith($HostedServerUrl)))
+                    {
+                        $serviceUrl = "$HostedServerUrl/rest/services/Utilities/Geometry/GeometryServer"
+                        $helperServiceParams.Add("geometryService",  '{"url": "' + $serviceUrl + '" }')
+                    }
+                } else {
+                    Write-Warning "No Hosted Server available. Geometry-Service is not working."
+                }
+            }
+            elseif ($HelperSrvcs.geometry.url)
```
Comment (shailesh91) [inline Modules/ArcGIS/DSCResources/ArcGIS_Portal_DisconnectedEnvironment/ArcGIS_Portal_DisconnectedEnvironment.psm1]: @spitzerr We will need another case for support for secure service which requires a credential. https://enterprise.arcgis.com/en/portal/latest/administer/linux/configure-services.htm#ESRI_SECTION2_CD97431BAC104D9A80DEA0D55F4420D9

**Comment by shailesh91** [comment]
Route Service needs to be published from disconnected service resource using the following utility (https://enterprise.arcgis.com/en/server/latest/administer/windows/publishing-routing-services.htm).

**Comment by mlewis22** [comment]
Has this progressed into build?

**Comment by shailesh91** [comment]
@mlewis22 this is on the roadmap but this will not be progressing into the build in 10.7 timeframes.

**Comment by mlewis22** [comment]
Thanks, i will keep an eye out

**Pull request closed without merging**